### PR TITLE
Fix 1117 [2/2] Replace old soa_vector with new one

### PR DESCRIPTION
**Pull request by jfalcou**

### CODE REVIEW

File: test/unit/algo/container/soa_vector.cpp [hunk 1]
```diff
@@ -203,73 +560,6 @@ TTS_CASE("Check soa_vector::pop behavior")
...
-  int i = 0;
-  while(tv_begin != tv_end)
-    TTS_EQUAL( eve::read(tv_begin++), tv.get(i++));
-
-  auto uv_begin = uv.begin();
-  auto uv_end   = uv.end();
-  i = 0;
-  while(uv_begin != uv_end)
-    TTS_EQUAL( eve::read(uv_begin++), uv.get(i++));
-};
```
Comment (DenisYaroshevskiy) [inline test/unit/algo/container/soa_vector.cpp:250]: Hey, this seems likee a good test!

File: test/unit/algo/container/soa_vector.cpp [hunk 2]
```diff
@@ -11,79 +11,382 @@
...
+  TTS_TYPE_IS(v::const_pointer_aligned, (eve::algo::views::zip_iterator<const_ap,   const_ap>));
+
+  TTS_TYPE_IS(v::iterator,               (eve::algo::views::converting_iterator<v::pointer,               T>));
+  TTS_TYPE_IS(v::const_iterator,         (eve::algo::views::converting_iterator<v::const_pointer,         T>));
+  TTS_TYPE_IS(v::iterator_aligned,       (eve::algo::views::converting_iterator<v::pointer_aligned,       T>));
+  TTS_TYPE_IS(v::const_iterator_aligned, (eve::algo::views::converting_iterator<v::const_pointer_aligned, T>));
+
+  TTS_CONSTEXPR_EXPECT((std::convertible_to<v::iterator, v::const_iterator>));
+  TTS_CONSTEXPR_EXPECT((std::convertible_to<v::pointer,  v::const_pointer>));
+};
```
Comment (DenisYaroshevskiy) [inline test/unit/algo/container/soa_vector.cpp:36]: Why comment out?

File: test/unit/algo/container/soa_vector.cpp [hunk 3]
```diff
@@ -203,73 +560,6 @@ TTS_CASE("Check soa_vector::pop behavior")
...
-  TTS_TYPE_IS(v::const_pointer_aligned, (eve::algo::views::zip_iterator<const_ap,   const_ap>));
-
-  TTS_TYPE_IS(v::iterator,               (eve::algo::views::converting_iterator<v::pointer,               T>));
-  TTS_TYPE_IS(v::const_iterator,         (eve::algo::views::converting_iterator<v::const_pointer,         T>));
-  TTS_TYPE_IS(v::iterator_aligned,       (eve::algo::views::converting_iterator<v::pointer_aligned,       T>));
-  TTS_TYPE_IS(v::const_iterator_aligned, (eve::algo::views::converting_iterator<v::const_pointer_aligned, T>));
-
-  TTS_CONSTEXPR_EXPECT((std::convertible_to<v::iterator, v::const_iterator>));
-  TTS_CONSTEXPR_EXPECT((std::convertible_to<v::pointer,  v::const_pointer>));
-};
```
Comment (DenisYaroshevskiy) [inline test/unit/algo/container/soa_vector.cpp:271]: That's a good test!

File: test/unit/algo/container/soa_vector.cpp (hunk 2 above)
Comment (jfalcou) [reply test/unit/algo/container/soa_vector.cpp:36]: Mistake

File: test/unit/algo/container/soa_vector.cpp (hunk 1 above)
Comment (jfalcou) [reply test/unit/algo/container/soa_vector.cpp:250]: It has been moved l 145 to be next to the begin/end test. See above.

File: test/unit/algo/container/soa_vector.cpp (hunk 3 above)
Comment (jfalcou) [reply test/unit/algo/container/soa_vector.cpp:271]: Same, moved first

**Pull request merged**